.github/workflows/inspektor-gadget.yml

```diff
@@ -233,32 +236,28 @@ jobs:
     - name: Login to Container Registry
       uses: docker/login-action@v1
       with:
-        registry: ${{ secrets.CONTAINER_REGISTRY }}
-        username: ${{ secrets.CONTAINER_REGISTRY_USERNAME }}
-        password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
     - name: Check out code
       uses: actions/checkout@v1
-    - name: Set IMAGE_TAG
-      run: |
-        TMP1=${GITHUB_REF#*/}
-        TMP2=${TMP1#*/}
-        IMAGE_TAG=${TMP2//\//-}
-        if [ "$IMAGE_TAG" = "main" ]; then
-            IMAGE_TAG="latest"
-        fi
-        echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
+    - name: Set environment variables.
+      uses: ./.github/actions/set-env
+      with:
+        # We will push the image to inspektor-gadget-ci before integration
+        # tests.
+        # If the tests are OK, the image will be pushed to inspektor-gadget
+        # below.
+        container-image: ${{ env.CONTAINER_IMAGE_CI }}
     - name: Build gadget container
       uses: docker/build-push-action@v2
       with:
         context: /home/runner/work/inspektor-gadget/inspektor-gadget/
         file: /home/runner/work/inspektor-gadget/inspektor-gadget/gadget.Dockerfile
         build-args: |
           ENABLE_BTFGEN=true
-        # TODO: how to avoid pushing a container before running integration tests
-        # Answer: push to runner registry first, if integration tests are OK
-        # push to final registry
         push: true
-        tags: ${{ secrets.CONTAINER_REPO }}:${{ env.IMAGE_TAG }}
+        tags: ${{ env.CONTAINER_REPO }}:${{ env.IMAGE_TAG }}
```

Comment:
While I agree setting env variables inside this action is strange at it hides the thing the truth is we are using env variables in a lot of places (basically each `Makefile`).
So, with converting this action to output, each time we will call make we will need to write something like this:

```bash
export IMAGE_TAG=${{ foo.output.image_tag }}
export CONTAINER_REPO=${{ foo.output.container_repo }}
make bar
```

I am not really convinced about adding the above snippet before each call to `make`.
For now, I will stick with the action as it is.

Reply:
It will remain like this for the moment.
We can modify it later if we do not like it.
Also, I actually split the action into one to set IMAGE_TAG and the other of CONTAINER_REPO.

Reply:
Also, the place you linked is the only place where we build using docker action and setting explicitly the tags.
There is another one when we use `docker` directly to "retag" the image.
But on the other places, we are using `make`, which uses `CONTAINER_REPO` and `IMAGE_TAG`, to build the CLI, the container image, run the tests and tag the test image.